### ISSUE TITLE
3.4.0-alpha.7 on react-native: require.resolve is not a function

### ISSUE THREAD

**Issue by macrozone** [storybooks/storybook#2943]
using 3.4.0-alpha.7 on react-native leads to this error on the app:

```
require.resolve is not a function. (In 'require.resolve('./dist/client/manager')', 'require.resolve' is undefined)

<unknown>
    client.js:5:31
loadModuleImplementation
    require.js:213:12
<unknown>
    index.js:39:14
loadModuleImplementation
    require.js:213:12
<unknown>
    index.js:16:15
loadModuleImplementation
    require.js:213:12
<unknown>
    storybook.js:5
loadModuleImplementation
    require.js:213:12
<unknown>
    index.js:1
loadModuleImplementation
    require.js:213:12
guardedLoadModule
    require.js:140:45
global code
    <unknown file>:0
```

**Comment by Hypnosphi** [comment]
Related: https://github.com/storybooks/storybook/pull/2788 #2796

**Comment by igor-dv** [comment]
Yeah... probably should be moved to `core/server.js`

**Comment by Hypnosphi** [comment]
Released as [v3.4.0-alpha.8](https://github.com/storybooks/storybook/releases/tag/v3.4.0-alpha.8)
